dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,276 @@
+package com.dotmarketing.business.cache.provider;
+
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.lang.reflect.Array;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UtilMethods;
+import com.github.benmanes.caffeine.base.UnsafeAccess;
+
+
+public class CacheSizingUtil {
+
+
+    final int standardSampleSize = Config.getIntProperty("CACHE_SIZER_SAMPLE_SIZE", 20);
+    
+    final int maxRecusionDepth = 10;
+    
+    final boolean useCompressedOops = true;
+    
+    public String averageSizePretty(final Map<String, Object> cacheMap) {
+
+        return UtilMethods.prettyByteify(averageSize(cacheMap));
+
+    }
+
+    /**
+     * Takes a map and pulls random values from it to calculate an average object size
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long averageSize(final Map<String, Object> cacheMap) {
+
+        if (cacheMap.size() == 0) {
```

Comment:
Codacy found an issue: [Substitute calls to size() == 0 (or size() != 0, size() > 0, size() < 1) with calls to isEmpty()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)